src/hotspot/cpu/aarch64/immediate_aarch64.cpp

```diff
@@ -295,7 +295,7 @@ static int expandLogicalImmediate(uint32_t immN, uint32_t immr,
     uint64_t and_bits_sub = replicate(and_bit, 1, nbits);
     uint64_t or_bits_sub = replicate(or_bit, 1, nbits);
     uint64_t and_bits_top = (and_bits_sub << nbits) | ones(nbits);
-    uint64_t or_bits_top = (0 << nbits) | or_bits_sub;
+    uint64_t or_bits_top = (0UL << nbits) | or_bits_sub;
```

Comment:
This should use UCONST64 instead of a UL suffix, because that suffix is platform-dependent. Windows is
LLP64, so L is 32 bits. This is code that is (probably, I haven't checked for sure) used by the windows-aarch64
port.